parsing/intermediate/dialects/defaultdialect/src/test/java/org/apache/calcite/test/DefaultDialectParserTest.java

```diff
@@ -27,4 +29,32 @@
   @Override protected SqlParserImplFactory parserImplFactory() {
     return DefaultDialectParserImpl.FACTORY;
   }
+
+  @Test
+  void testParensInFrom() {
+    // UNNEST may not occur within parentheses.
+    // FIXME should fail at "unnest"
+    sql("select *from ^(^unnest(x))")
+        .fails("(?s)Encountered \"\\( unnest\" at .*");
+
+    // <table-name> may not occur within parentheses.
+    sql("select * from (^emp^)")
+        .fails("(?s)Non-query expression encountered in illegal context.*");
+
+    // <table-name> may not occur within parentheses.
+    sql("select * from (^emp^ as x)")
+        .fails("(?s)Non-query expression encountered in illegal context.*");
+
+    // <table-name> may not occur within parentheses.
+    sql("select * from (^emp^) as x")
+        .fails("(?s)Non-query expression encountered in illegal context.*");
+
+    // Parentheses around JOINs are OK, and sometimes necessary.
+    if (false) {
```

Comment:
Since it has a todo, I erred on the side of leaving it in case someone actually decides to come back after all these years and change it (not super likely, but you never know...).

Reply:
+1 for removing it. At this point, it's dead code :)

Reply:
Ok, SGTM